src/memory_map.rs

```diff
@@ -1,5 +1,15 @@
 use core::marker::PhantomData;
 
+/// This Tag provides an initial host memory map.
+///
+/// The map provided is guaranteed to list all standard RAM that should be
+/// available for normal use. This type however includes the regions occupied
+/// by kernel, mbi, segments and modules. Kernel must take care not to
+/// overwrite these regions.
+///
+/// This tag may not be provided by some boot loaders on EFI platforms if EFI
+/// boot services are enabled and available for the loaded image
+/// (EFI boot services not terminated tag exists in Multiboot2 information structure).   
```

Comment:
Right, thats another copy paste failure. This time from the [GNU GRUB multiboo2 refrence](https://www.gnu.org/software/grub/manual/multiboot2/multiboot.html#Memory-map).

Essentially it states that the memory map tag may not be provided on EFI platforms if EFI boot services are enabled, and then notes that there is [an EFI boot services tag](https://www.gnu.org/software/grub/manual/multiboot2/multiboot.html#EFI-boot-services-tag) in the standard.

I've cleaned it up so that the explanation comes across more clearly.